### DISCUSSION TITLE
zmq mode port configuration

### DISCUSSION THREAD

**Discussion by Zplusless** [srsran/srsRAN_Project#406] [General Help] [answered]
Hi, I'm using srsRAN with its zmq mode.  But I have some problems on how to configure zmq mode:

1. In the tutorial of [srsRAN gNB with Amarisoft UE](https://docs.srsran.com/projects/project/en/latest/tutorials/source/amariUE/source/index.html), the gnb is configured with ` tx_port=tcp://10.53.1.1:5000,rx_port=tcp://10.53.1.2:6000` and the Amarisoft UE is configured with `tx_port0:  "tcp://10.53.1.2:6000",
  rx_port0:  "tcp://10.53.1.1:5000",`. However, which ip is gnb's and which ip is ue's is not given. Is it correct that using its own ip to configure rx and the peer's ip to configure tx? In this tutorial, 10.53.1.1 is ue and 10.53.1.2 is gnb?

2. If the above speculation is correct, can srsRAN+srsUE be configured following the same configure for zmq mode (ue and gnb run in different server)?

**Comment by herlesupreeth** [comment, marked as answer]
In gNB the IP in `tx_port=` is the gNB IP and the IP in `rx_port=` is the UE IP

**Comment by Zplusless** [comment]
> In gNB the IP in `tx_port=` is the gNB IP and the IP in `rx_port=` is the UE IP

Thanks！
